### ISSUE TITLE
Very low stop loss giving spurious backtest results

### ISSUE THREAD

**Issue by zlac116** [freqtrade/freqtrade#3841]
I have setup a strategy which looks for substantial drops in the market over 2-5 candles on a 1hour timeframe. The stoploss is set to ~1% with a trailing stoploss per below. The backtests seems to perform remarkably well but the dry and live runs don't seem to perform in line with them. Below I have highlighted the strategy file and I can't seem to find any subtle errors that might affect the backtests results. However, if I increase the stoploss to ~3%, I seem to get more realistic backtest results. I'd appreciate any help/guidance that you can give here? many thanks

**Strategy file**

```
class Strategy1hBot1(IStrategy):
    """
    Strategy 002
    author@: Gerald Lonlas
    github@: https://github.com/freqtrade/freqtrade-strategies
    How to use it?
    > python3 ./freqtrade/main.py -s Strategy002
    """
    minimal_roi = {
        "0": 0.55172,
        "384": 0.09632,
        "1098": 0.05352,
        "2467": 0
    }

    # Optimal stoploss designed for the strategy
    # This attribute will be overridden if the config file contains "stoploss"
    stoploss = -0.01002

    # Optimal ticker interval for the strategy
    ticker_interval = '1h'

    # Run "populate_indicators()" only for new candle.
    process_only_new_candles = False

    # trailing stoploss
    trailing_only_offset_is_reached = False
    trailing_stop = True
    trailing_stop_positive = 0.31219
    trailing_stop_positive_offset = 0.36459

    #Optional order type mapping
    order_types = {
        'buy': 'limit',
        'sell': 'limit',
        'stoploss': 'market',
        'stoploss_on_exchange': False
    }
    plot_config = {
        # Main plot indicators (Moving averages, ...)
        'main_plot': {
            # 'tema': {},
            # 'bb_lowerband1.0': {'color': 'green'},
            # 'bb_middleband': {'color': 'green'},
            # 'bb_upperband2.0': {'color': 'green'},
            # 'ema50': {'color': 'blue'},
            # }
        },
        'subplots': {
            # Subplots - each dict defines one additional plot
            "MACD": {
                'macd': {'color': 'blue'},
                'macdsignal': {'color': 'orange'},
            },
            "RSI-14": {
                'rsi14': {'color': 'red'},
            },
            "RSI-2": {
                'rsi2': {'color': 'black'},
            },
            "BB Percent": {
                'bb_percent': {'color': 'blue'},
            }
        }
    }

    def informative_pairs(self):
        """
        Define additional, informative pair/interval combinations to be cached from the exchange.
        These pair/interval combinations are non-tradeable, unless they are part
        of the whitelist as well.
        For more information, please consult the documentation
        :return: List of tuples in the format (pair, interval)
            Sample: return [("ETH/USDT", "5m"),
                            ("BTC/USDT", "15m"),
                            ]
        """
        return []
    def populate_indicators(self, dataframe: DataFrame) -> DataFrame:

        # RSI
        dataframe['rsi14'] = ta.RSI(dataframe, timeperiod=14)
        dataframe['rsi3'] = ta.RSI(dataframe, timeperiod=3)

        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['rsi3'] < 35) &
                (
                    (dataframe['close'] / dataframe['close'].shift(1) < 0.96536) | 
                    (dataframe['close'] / dataframe['close'].shift(2) < 0.92987) |
                    (dataframe['close'] / dataframe['close'].shift(3) < 0.92166) |
                    (dataframe['close'] / dataframe['close'].shift(4) < 0.93387) |
                    (dataframe['close'] / dataframe['close'].shift(5) < 0.90575) 
                ) &
                (dataframe['volume'] > 0)
            ),
            'buy'] = 1
        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['rsi14'] > 56) &
                (dataframe['rsi3'] > 60) &
                (dataframe['volume'] > 0)
            ),
            'sell'] = 1
        return dataframe
```

**Comment by xmatthias** [comment]
Please carefully read and understand and think through the [Backtesting assumptions](https://www.freqtrade.io/en/latest/backtesting/#assumptions-made-by-backtesting) documentation.

The answer is right there (specifically in how trailing stoploss is handled).

If you still have questions after that don't hesitate to ask more detailed questions.

**Comment by zlac116** [comment]
Thanks @xmatthias. I missed this excerpt in the docs. Seems to me that when I am trying to optimise the strategy, using the hyperopt function, the backtest is exploiting the gains through the `trailing_stop`.

**Doc excerpt**

Stoploss (and trailing stoploss) is evaluated before ROI within one candle. So you can often see more trades with the stoploss and/or trailing_stop sell reason comparing to the results obtained with the same strategy in the Dry Run/Live Trade modes.

**Comment by zlac116** [comment]
Aswell as the use of highs and lows per candle for the stoploss and trailing stops, meaning that the stoploss that I have in my strategy is too low to get a meaningful backtest. Thanks @xmatthias

**Comment by xmatthias** [comment]
Exactly right :+1: 

it's a problem, yes - but as it's unknown what happens within a candle, certain assumptions have to be taken.
We try to make sensible assumptions - but the devil is always in the detail ...
and tight stoplosses are unfortunately a very known problem.

**Comment by zlac116** [comment]
Thanks @xmatthias. Agreed and I think that the assumptions about what happens between candles are sensible enough for backtests, given the data feeding the strategies.
